ocaml/xenopsd/xc/xenops_server_xen.ml

```diff
@@ -1280,12 +1280,13 @@ module VM = struct
 
   let mkints n = List.init n Fun.id
 
-  let generate_create_info ~xc ~xs:_ vm persistent =
+  let generate_create_info ~xs:_ vm persistent =
     let ty = match persistent.VmExtra.ty with Some ty -> ty | None -> vm.ty in
     let hvm = match ty with HVM _ | PVinPVH _ -> true | PV _ -> false in
     (* XXX add per-vcpu information to the platform data *)
     (* VCPU configuration *)
-    let pcpus = Xenctrlext.get_max_nr_cpus xc in
+    let xcext = Xenctrlext.get_handle () in
+    let pcpus = Xenctrlext.get_max_nr_cpus xcext in
```

Comment:
Now (almost) every call to a `Xenctrlext` comes with an additional `Xenctrlext.get_handle ()` call just above. With the handle being cached indefinitely upon the first call, then you might as well call `get_handle ()` inside the library itself as part of every library call.